src/webgldrawer.js

```diff
@@ -276,10 +276,15 @@
                 let maxTextures = this._gl.getParameter(this._gl.MAX_TEXTURE_IMAGE_UNITS);
                 if(maxTextures <= 0){
                     // This can apparently happen on some systems if too many WebGL contexts have been created
-                    // in which case maxTextures can be null, leading to out of bounds errors with the array
+                    // in which case maxTextures can be null, leading to out of bounds errors with the array.
+                    // For example, when viewers were created and not destroyed in the test suite, this error
+                    // occured in the TravisCI tests, though it did not happen when testing locally either in
+                    // a browser or on the command line via grunt test.
+
                     // use plain console.error instead of $.console.error in order to have the message show up in the test log.
                     console.error(`There was a WebGL problem: bad value for MAX_TEXTURE_IMAGE_UNITS (${maxTextures})`);
-                    return;
+                    throw(new Error(`WegGL error: bad value for gl parameter MAX_TEXTURE_IMAGE_UNITS (${maxTextures}). This could happen
```

Comment:
Do we need both `console.error` and `throw`? Also, what's the end result of this throw? What does the user see? Is there a better way for us to limp along or is it good that we just fail, since there's not much we can do at this stage? 

I suppose one aggressive thing we could do would be to switch to the canvas drawer and continue with that? That's probably a decision better left to the developer, though. At any rate, hopefully a dramatic error like this will cause the developer to fix their code!

Reply:
I understand we need the `throw` for the testing side... Just making sure it's also appropriate for the use side.

Reply:
Yeah, we probably don't need the console message anymore since throwing the error should presumably lead to a message being printed the console also.

I really don't know when this would happen. The only time I've seen it is in the Travis tests - never in a browser. The webgl spec says the minimum value should be 8 and doesn't mention anything about the possibility that it could be `null.` I don't think switching automatically to `CanvasDrawer` is necessarily always a good idea - it would break any features that depend on `webgl`- and perhaps `webgl` would start working again on a subsequent frame. Again though, I'm not sure if this will ever happen in practice in a browser that we support.